crates/amaru-ledger/src/rules/context.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use amaru_kernel::{
+    Anchor, CertificatePointer, DRep, Lovelace, PoolId, PoolParams, StakeCredential,
+    TransactionInput, TransactionOutput,
+};
+
+pub use simple::*;
+pub(crate) mod fake;
+mod simple;
+
+/// The BlockValidationContext is a collection of slices needed to validate a block
+pub trait BlockValidationContext:
+    PotsSlice + UtxoSlice + PoolsSlice + AccountsSlice + DRepsSlice
+{
+}
+
+/// The BlockPreparationContext is a collection of interfaces needed to prepare a block
+pub trait BlockPreparationContext<'a>:
+    PrepareUtxoSlice<'a> + PreparePoolsSlice<'a> + PrepareAccountsSlice<'a> + PrepareDRepsSlice<'a>
+{
+}
```

Comment:
When using these small slices it is usually nice to add a blanket implementation for anything that implements all the slices.

Reply:
What would that look like, and why is it nice? (genuine question)

Reply:
```rs
impl<'a, T> BlockPreparationContext<'a> for T
where T: PrepareUtxoSlice<'a> + PreparePoolsSlice<'a> + ...
{}
```